test/Microsoft.IdentityModel.Tokens.Tests/CryptoProviderFactoryTests.cs

```diff
@@ -1184,23 +1187,22 @@ public void ProviderCache_EnsureNoLeakingTasks_SecurityTokenHandler_SigningCrede
             var handler = new JwtSecurityTokenHandler();
             _ = handler.WriteToken(token);
 
-            var cache = CryptoProviderFactory.Default.CryptoProviderCache as InMemoryCryptoProviderCache;
-            var waitTimeInSeconds = TaskStopWaitTimeInSeconds(cache.TaskExecutionTimeInSeconds);
-
-            WaitTillTasksStarted(cache, waitTimeInSeconds); // wait for the event queue task to start
-            Assert.True(cache.TaskCount == 1, $"ProviderCache_EnsureNoLeakingTasks_SecurityTokenHandler_SymmetricEncryptingCredentials: unexpected task count: {cache.TaskCount}, expected: 1");
+            AssertNoHangingingTasks(CryptoProviderFactory.Default.CryptoProviderCache as InMemoryCryptoProviderCache, "ProviderCache_EnsureNoLeakingTasks_SecurityTokenHandler_SigningCredentials");
+        }
 
-            WaitTillTasksComplete(cache, waitTimeInSeconds); // wait for the event queue task to complete
-            Assert.True(cache.TaskCount == 0, $"ProviderCache_EnsureNoLeakingTasks_SecurityTokenHandler_SigningCredentials: unexpected task count: {cache.TaskCount}, expected: 0");
+        private void AssertNoHangingingTasks(InMemoryCryptoProviderCache cache, string callName)
+        {
+            WaitTillTasksComplete(cache, TaskStopWaitTimeInSeconds(cache.TaskExecutionTimeInSeconds)); // wait for the event queue task to complete
```

Comment:
>WaitTillTasksComplete [](http://example.com/codeflow?start=12&length=21)

how much time is pend waiting in our utests?

Reply:
Currently, the event queue task will end after "Last_Event Time" + 2 min, but in tests the wait is 1.5 times (3 min) to accommodate any overhead.